### DISCUSSION TITLE
how to get logit from rnninfer

### DISCUSSION THREAD

**Discussion by taylorchu** [NVIDIA-NeMo/NeMo#4503] [Q&A] [answered]
Do you have some examples of extracting raw logits from rnninfer(https://github.com/NVIDIA/NeMo/blob/0af1431499e57caf91cee33c2593465630d44d09/nemo/collections/asr/parts/submodules/rnnt_greedy_decoding.py#L369)?

Thanks!

**Comment by titu1994** [comment, marked as answer]
We don't support that as of now. Unlike CTC, RNNT log probs are computed autoregressively and therefore we can't return the full joint. We maybe could consider adding a flag to return the decoded joint as a dangling list in a future version. 

If you want only the label ids, you can do that with changing the decoding strategy to preserve alignments. The docs and tutorials have explanation of how to do that.

**Reply by taylorchu** [reply]
I am trying to add "custom word boost" with external beam search library like pyctcdecode, so I need logprob for each token.
RNNT infer is much more accurate than ctc version, probably because of RNN decoder (while CTC version has several misspellings). so preferably, I want to keep using the same RNN decoder model. Maybe there is already a possible way to boost words with just nemo.

**Reply by titu1994** [reply]
This is very interesting, and something we were considering to add (but there's been other priorities). 

For the time being, you can probably modify the class locally so that af any place that has the https://github.com/NVIDIA/NeMo/blob/0af1431499e57caf91cee33c2593465630d44d09/nemo/collections/asr/parts/submodules/rnnt_greedy_decoding.py#L482 flag called preserve_alignments, instead of preserving the label, you can preserve the logprobs (but remember to .detach()). 

Currently SLT is taking up our priority but I will try to add proper support for this on main branch soon. 

Fyi @VahidooX
